Tutorials/WinForms/Classic/bind-the-data-grid-to-large-data-source.md

```diff
@@ -0,0 +1,294 @@
+                Form form = null;
+                switch (e.Element.Text) {
+                    case OrdersFormName:
+                        form = ordersForm;
+                        break;
+                    case CustomersFormName:
+                        form = customersForm;
+                        break;
+                }
+                if (form != null) {
+                    tabbedView.AddDocument(form);
+                    tabbedView.ActivateDocument(form);
+                }
+            }
+            void barButtonNavigation_ItemClick(object sender, ItemClickEventArgs e) {
+                SetAccordionSelectedElement(e.Item.Caption);
+            }
+            void tabbedView_DocumentClosed(object sender, DocumentEventArgs e) {
+                RecreateForms(e);
+            }
+            void SetAccordionSelectedElement(string name) {
+                switch (name) {
+                    case OrdersFormName:
+                        accordionControl.SelectedElement = ordersAccordionControlElement;
+                        break;
+                    case CustomersFormName:
+                        accordionControl.SelectedElement = customersAccordionControlElement;
+                        break;
+                }
+            }
+            void RecreateForms(DocumentEventArgs e) {
+                switch (e.Document.Caption) {
+                    case OrdersFormName:
+                        ordersForm = CreateForm(OrdersFormName);
+                        break;
+                    case CustomersFormName:
+                        customersForm = CreateForm(CustomersFormName);
+                        break;
+                }
+            }
+        }
+    }
+    ```
+    </details>
+
+* For C# projects only. Open the **Program.cs** file, and change the following line
+    ```csharp
+    Application.Run(new CustomersListForm());
+    ```
+    to
+    ```csharp
+    Application.Run(new MainForm());
+    ```
+* Form VB.NET projects only. 
+    * Right-click the project item in the Solution Explorer window to invoke the context menu and select the **Properties** menu item (you can use the **Alt+Enter** keyboard shortcut instead). In the Properties window select the **Application** page and change the **Startup form** property to **MainForm**
+    * Open the **CustomersListForm** code, cut the **ConnectionHelper** and **DemoDataHelper** code and paste it to the **MainForm** constructor
+        ```vbnet
+        Public Sub New()
+            ConnectionHelper.Connect(False)
+            Using uow As New UnitOfWork()
+                DemoDataHelper.Seed(uow)
+            End Using
+            InitializeComponent()
+            ordersForm = CreateForm(OrdersFormName)
+            customersForm = CreateForm(CustomersFormName)
+            accordionControl.SelectedElement = ordersAccordionControlElement
+        End Sub
+        ```
+* Run the application and click the **Orders** navigation item to see the result
```

Comment:
```suggestion
* Run the application and click the **Orders** navigation item to see the result.
```

Reply:
I'll check the rest after you thoroughly go through this entire PR and correct all the issues I've highlighted.

Reply:
Done!